shared/js/baseOptions.js

```diff
@@ -0,0 +1,149 @@
+startTiming('Puzzle', script_path, '');
```

Comment:
What about `initApp.js`?  The contents of this file haven't been refactored yet, this change is just to get the high level pieces in approx. the right places.